launchdarkly-android-client-sdk/src/androidTest/java/com/launchdarkly/sdk/android/LDClientEventTest.java

```diff
@@ -196,6 +196,48 @@ public void variationFlagTrackReasonGeneratesEventWithReason() throws IOExceptio
         }
     }
 
+    @Test
+    public void flagEvaluationWithPrereqProducesPrereqEvents() throws IOException, InterruptedException {
+        try (MockWebServer mockEventsServer = new MockWebServer()) {
+            mockEventsServer.start();
+            // Enqueue a successful empty response
+            mockEventsServer.enqueue(new MockResponse());
+
+            // Setup flag store with test flag
+            Flag flagA = new FlagBuilder("flagA").version(1)
+                    .variation(1).value(LDValue.of(true)).reason(EvaluationReason.targetMatch()).build();
+            Flag flagAB = new FlagBuilder("flagAB").prerequisites(new String[]{"flagA"}).version(1)
+                    .variation(1).value(LDValue.of(true)).reason(EvaluationReason.targetMatch()).build();
+            Flag flagAC = new FlagBuilder("flagAC").prerequisites(new String[]{"flagA"}).version(1)
+                    .variation(1).value(LDValue.of(true)).reason(EvaluationReason.targetMatch()).build();
+            Flag flagABD = new FlagBuilder("flagABD").prerequisites(new String[]{"flagAB"}).version(1)
+                    .variation(1).value(LDValue.of(true)).reason(EvaluationReason.targetMatch()).build();
+            PersistentDataStore store = new InMemoryPersistentDataStore();
+            TestUtil.writeFlagUpdateToStore(store, mobileKey, ldContext, flagA);
+            TestUtil.writeFlagUpdateToStore(store, mobileKey, ldContext, flagAB);
+            TestUtil.writeFlagUpdateToStore(store, mobileKey, ldContext, flagAC);
+            TestUtil.writeFlagUpdateToStore(store, mobileKey, ldContext, flagABD);
+            LDConfig ldConfig = baseConfigBuilder(mockEventsServer)
+                    .persistentDataStore(store).build();
+
+            try (LDClient client = LDClient.init(application, ldConfig, ldContext, 0)) {
+                assertTrue(client.boolVariation("flagA", false));
+                assertTrue(client.boolVariation("flagAB", false));
+                assertTrue(client.boolVariation("flagAC", false));
+                assertTrue(client.boolVariation("flagABD", false));
+                client.blockingFlush();
+
+                LDValue[] events = getEventsFromLastRequest(mockEventsServer, 2);
+                LDValue summaryEvent = events[1];
+                assertSummaryEvent(summaryEvent);
+                assertEquals(LDValue.of(4), summaryEvent.get("features").get("flagA").get("counters").get(0).get("count"));
```

Comment:
`summaryEvent.get("features").get("flagA").get("counters").get(0).get("count")`
Cires for the law of demeter.

(I don't actually want any changes here.)

Reply:
Yes, this is gross, but is test.